packages/core/contracts/oracle/implementation/Staker.sol

```diff
@@ -77,6 +129,13 @@ contract Staker is StakerInterface, Ownable, Testable {
         voterStakes[msg.sender].pendingUnstake = amount;
         voterStakes[msg.sender].activeStake -= amount;
         voterStakes[msg.sender].unstakeTime = getCurrentTime() + unstakeCoolDown;
+        emit RequestedUnstake(
```

Comment:
it might be better for this method to define a `VoterStake storage voterStake = voterStakes[msg.sender];` and then reference this. you can leave it for now but just looking at some of the changes you have here

Reply:
Well remarked, I finally decided to focus on events and leave the storage reading optimisations for a second pass. But there is definitely room for improvement here.